File: app/src/main/java/com/johnnyzhou/movieme/di/component/AppComponent.java
package com.johnnyzhou.movieme.di.component;

import android.app.Application;

import com.johnnyzhou.movieme.MovieMeApplication;
import com.johnnyzhou.movieme.di.module.ApiModule;
import com.johnnyzhou.movieme.di.module.AppModule;
import com.johnnyzhou.movieme.network.NetworkManager;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Component;
import de.greenrobot.event.EventBus;
import rx.Scheduler;

@Singleton
@Component(modules = {AppModule.class, ApiModule.class})
public interface AppComponent {
    void inject(MovieMeApplication application);

    NetworkManager networkManager();

    Application application();

    @Named(AppModule.MAIN_THREAD)
    Scheduler mainThread();

    @Named(AppModule.IO_THREAD)
    Scheduler ioThread();

    EventBus Bus();
}

File: app/src/main/java/com/johnnyzhou/movieme/di/component/AppComponent.kt
package com.johnnyzhou.movieme.di.component

import android.app.Application
import com.johnnyzhou.movieme.MovieMeApplication
import com.johnnyzhou.movieme.business.DateProvider
import com.johnnyzhou.movieme.di.module.ApiModule
import com.johnnyzhou.movieme.di.module.AppModule
import com.johnnyzhou.movieme.network.MovieService
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class, ApiModule::class])
interface AppComponent {
    fun inject(application: MovieMeApplication)

    fun application(): Application

    val movieService: MovieService

    val dateProvider: DateProvider
}